cmd/xgql/main.go

```diff
@@ -87,15 +90,16 @@ var noCache = []client.Object{
 
 func main() {
 	var (
-		app      = kingpin.New(filepath.Base(os.Args[0]), "A GraphQL API for Crossplane.").DefaultEnvars()
-		debug    = app.Flag("debug", "Enable debug logging.").Short('d').Bool()
-		listen   = app.Flag("listen", "Address at which to listen for TLS connections. Requires TLS cert and key.").Default(":8443").String()
-		tlsCert  = app.Flag("tls-cert", "Path to the TLS certificate file used to serve TLS connections.").ExistingFile()
-		tlsKey   = app.Flag("tls-key", "Path to the TLS key file used to serve TLS connections.").ExistingFile()
-		insecure = app.Flag("listen-insecure", "Address at which to listen for insecure connections.").Default("127.0.0.1:8080").String()
-		play     = app.Flag("enable-playground", "Serve a GraphQL Playground.").Bool()
-		agent    = app.Flag("trace-agent", "Address of the Jaeger trace agent. Leave unset to disable tracing.").String()
-		ratio    = app.Flag("trace-ratio", "Ratio of queries that should be traced.").Default("0.01").Float()
+		app         = kingpin.New(filepath.Base(os.Args[0]), "A GraphQL API for Crossplane.").DefaultEnvars()
+		debug       = app.Flag("debug", "Enable debug logging.").Short('d').Bool()
+		listen      = app.Flag("listen", "Address at which to listen for TLS connections. Requires TLS cert and key.").Default(":8443").String()
+		tlsCert     = app.Flag("tls-cert", "Path to the TLS certificate file used to serve TLS connections.").ExistingFile()
+		tlsKey      = app.Flag("tls-key", "Path to the TLS key file used to serve TLS connections.").ExistingFile()
+		insecure    = app.Flag("listen-insecure", "Address at which to listen for insecure connections.").Default("127.0.0.1:8080").String()
+		play        = app.Flag("enable-playground", "Serve a GraphQL Playground.").Bool()
+		tracer      = app.Flag("tracer", "Tracer to use. Set one of the following to enable tracing: \"jaeger\" or \"gcp\"").String()
+		ratio       = app.Flag("trace-ratio", "Ratio of queries that should be traced.").Default("0.01").Float()
+		jaegerAgent = app.Flag("jaeger-agent", "Address of the Jaeger trace agent as [host]:[port]").String()
```

Comment:
Will this be a breaking change for deployments that have `agent` configured?

Reply:
Yeah, good catch. 
I don't believe there are plenty of deployments (yet) where a jaeger tracing configured. 
However, it looks straightforward to handle this in a nonbreaking way with the overhead of a couple of lines of extra code.

If `--trace-agent` set:
- Log a deprecation warning
- Set tracer as "jaeger"
- Use it as `--jaeger-agent`

Reply:
Can we just leave this as `--trace-agent`? I could imagine it being potentially reused for other flavours of agent in future if we were to add any.

My suggestion would be to:

1. Have the `--tracer` / `--trace-backend` flag default to `jaeger`.
2. Enable Jaeger tracing if `--trace-backend` is `jaeger` && `--trace-agent` is set.

I believe this would maintain the current semantics while also allowing folks to enable GCP tracing (e.g. by setting `--trace-backend` to `gcp`, which would cause us to ignore `trace-agent`.

Reply:
Sounds good to me. 
One minor UX issue though, users could only set `--trace-backend=jaeger` and expect jaeger tracing to be enabled whereas we would be silently skipping enabling it without letting them know `--trace-agent` has to be set as well.

Reply:
Yeah, that's not ideal but I'm still prefer this approach since it maintains backward compatibility without needing to worry about deprecation and flag changes. We could include a note in the `--trace-agent` flag that it was required for jaeger.